acceptance/lib/facter/acceptance/base_fact_utils.rb

```diff
@@ -0,0 +1,613 @@
+module Facter
+  module Acceptance
+    module BaseFactUtils
+
+      # return the value from the facter json parsed set of results using the fact path separated using '.'s
+      #
+      def json_result_fact_by_key_path(results, fact)
+        fact.split('.').each do |key|
+          results = results[key]
+        end
+        results
+      end
+
+      # return a hash of expected facts for os, processors, and kernel facts
+      #
+      def os_processors_and_kernel_expected_facts(agent)
+        if agent['platform'] =~ /aix-/
+          aix_expected_facts(agent)
+        elsif agent['platform'] =~ /cisco/
+          cisco_expected_facts(agent)
+        elsif agent['platform'] =~ /cumulus-/
+          cumulus_expected_facts(agent)
+        elsif agent['platform'] =~ /debian-/
+          debian_expected_facts(agent)
+        elsif agent['platform'] =~ /el-|centos-/
+          el_expected_facts(agent)
+        elsif agent['platform'] =~ /eos-/ # arista
+          eos_expected_facts(agent)
+        elsif agent['platform'] =~ /fedora-/
+          fedora_expected_facts(agent)
+        elsif agent['platform'] =~ /huaweios-/
+          huawei_expected_facts(agent)
+        elsif agent['platform'] =~ /osx-/
+          osx_expected_facts(agent)
+        elsif agent['platform'] =~ /sles-/
+          sles_expected_facts(agent)
+        elsif agent['platform'] =~ /solaris-/
+          solaris_expected_facts(agent)
+        elsif agent['platform'] =~ /ubuntu-/
+          ubuntu_expected_facts(agent)
+        elsif agent['platform'] =~ /windows-/
+          windows_expected_facts(agent)
+        else
+          fail_test("unknown agent type being tested #{agent['platform']}")
+        end
+      end
+
+      # AIX
+      def aix_expected_facts(agent)
+        case agent['platform']
+          when /aix-5.3/
```

Comment:
If you don't want to have to update per release then we should collect the real value and test  it.